### PR TITLE
Create story

### LINKED ISSUE #7: Create a Story
As a refugee, I can create a Story to share my experience.

Create a route `/stories` and a database Model for a `Story`.
Only logged in Refugees are authorized to access this route.
On receiving a POST request on this route create a new `Story` with the fields `title` and `content`.
Validate that both fields are not empty and don't exceed a certain amount of characters. If they do return an appropriate error response and don't create the Story. 
When valid create the Story and return a 200 response with the Story details in the content.

### PR DESCRIPTION
**Pull request by Eltayeb-Elgaali**
closes #7

### CODE REVIEW

File: client/.eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...rc/Secret.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/Login.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/Header.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/EyeCatcher.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/Gallery.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/Logo.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/Navigation.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/Story.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/SmallStory.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/StoryBody.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/StoryFooter.jsx",[],"/home/hackyourfuture/Downloads/hack-my-future/Hear-My-Story/client/src/StoryHeader.jsx",[],{"ruleId":"71","severity":1,"message":"72","line":2,"column":10,"nodeType":"73","messageId":"74","endLine":2,"endColumn":14},"no-unused-vars","'Link' is defined but never used.","Identifier","unusedVar"]
```
Comment (Toinne) [inline client/.eslintcache]: Would remove this file should not be committed
https://github.com/Sayed94h/Hear-My-Story/pull/21/files Sayed added it to be ignored in the build fix PR

File: api/models/Story.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
 const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
 
 const StorySchema = new mongoose.Schema({
```
Comment (abelRoland) [inline api/models/Story.js:5]: Here it's also possible to use the variable `Schema` ;)

File: client/.eslintcache (hunk 1 above)
Comment (abelRoland) [inline client/.eslintcache]: Is this file needed?

File: api/controllers/stories.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const Story = require('../models/Story');
+const User = require('../models/User')
+
+
+module.exports.createStory =  async (req, res, next) => {
+    
+    if(!req.body.story) return res.status(400).send('No text provided');
```
Comment (Toinne) [inline api/controllers/stories.js]: Requirement: can you make sure the error follow the pattern of the other errors ?
```suggestion
    if (!req.body.story) {
       return res.status(400).send({error: true, message: "No story submitted"});
    }
```

File: api/controllers/stories.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    const story = new Story(req.body);
+    let token = req.cookies.auth;
+    
+    User.findByToken(token, (err, user) => {
+        if(err) {
+            return res(err)
+        }
+        if(user) {
+            story.author = user._id;
+            console.log(story);
```
Comment (Toinne) [inline api/controllers/stories.js]: Requirement: remove me

**Review by Toinne — CHANGES_REQUESTED** [review]

**Review by abelRoland — COMMENTED** [review]
Together with what @Toinne told, I have just some other small comments. Thanks for the work @Eltayeb-Elgaali

File: client/.eslintcache (hunk 1 above)
Comment (Sayed94h) [reply client/.eslintcache]: If you mean "the file .eslintcache", then I don't think that it is needed. I searched a lot to get rid of it but got nothing. That is why I added it in .gitignore file in the previous PR.

**Review by Sayed94h — COMMENTED** [review]
@Eltayeb-Elgaali 

Can you remove all the prefixes "cross-env" from [here](https://github.com/Sayed94h/Hear-My-Story/blob/main/client/package.json) and also this `"devDependencies": { "cross-env": "^7.0.3"	}` from the same file? because it caused failure in the deployment of our app on Heroku otherwise I will do that in a separate PR.
Thanks

File: api/controllers/stories.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
```
Comment (Toinne) [inline api/controllers/stories.js]:
```suggestion
```

**Review by Toinne — APPROVED** [review]

**Pull request merged**